pkg/git/internal/shared.go

```diff
@@ -0,0 +1,22 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/git/errors"
+)
+
+func HandleOutput(out string, err error) (string, error) {
+	lowerOut := strings.ToLower(out)
+	switch {
+	case err == nil:
+		return strings.TrimSpace(out), nil
+	case strings.Contains(lowerOut, "not a git repository"):
+		return "", errors.ErrNotARepository
+	case strings.Contains(lowerOut, "remote not found"):
+		return "", errors.ErrRemoteNotFound
+	default:
+		return "", fmt.Errorf("%s: %w", out, err)
```

Comment:
Odd: This error has a different format, it includes `out`.

Reply:
This is the only place where the output has any relevance